scenario.py

```diff
@@ -22,8 +18,8 @@ def __init__(self, is_quick_demo=False):
       
     self.dataset_name = 'MNIST'
     (x_train, y_train), (x_test, y_test) = mnist.load_data()    
-    self.x_train = x_train
-    self.y_train = y_train
+    # The train set has to be split into a train set and a validation set for early stopping (called 'esval' below)
+    self.x_train, self.x_esval, self.y_train, self.y_esval = train_test_split(x_train, y_train, test_size=0.2, random_state=42)
```

Comment:
Validation sets are usually referred as `x_val `and `y_val` not `..._esval`

Reply:
"esval" was for "Early Stopping validation", and prevent confusion with the `x_val` and `y_val` in the intra-node fitting. Can we keep that way for the moment?